demisto_sdk/commands/download/downloader.py

```diff
@@ -160,271 +140,312 @@ def __init__(
-                    string_to_write, scripts_id_to_name, Path(file_name).suffix
-                )
-            content_items_file_names_tuple.append((string_to_write, file.name))
-        return content_items_file_names_tuple, scripts_id_to_name
+        logger.info("Custom content IDs mapping created successfully.")
+        return mapping
 
-    def fetch_custom_content(self) -> bool:
+    def download_custom_content(self) -> dict[str, StringIO]:
         """
-        Fetches the custom content from Demisto into a temporary dir.
-        :return: True if fetched successfully, False otherwise
+        Download custom content bundle using server's API,
+        and create a StringIO object containing file data for each file within it.
+
+        Returns:
+            dict[str, StringIO]: A dictionary mapping custom content's file names to file objects.
         """
+        # Set to 'verify' to None so that 'demisto_client' will use the environment variable 'DEMISTO_VERIFY_SSL'.
+        verify = not self.insecure if self.insecure else None
+        logger.info("Fetching custom content bundle from server...")
+
         try:
-            verify = (
-                (not self.insecure) if self.insecure else None
-            )  # set to None so demisto_client will use env var DEMISTO_VERIFY_SSL
             self.client = demisto_client.configure(verify_ssl=verify)
-            api_response: tuple = demisto_client.generic_request_func(
-                self.client, "/content/bundle", "GET"
-            )
-            body: bytes = ast.literal_eval(api_response[0])
-            io_bytes = io.BytesIO(body)
-
-            # Demisto's custom content file is of type tar.gz
-            tar = tarfile.open(fileobj=io_bytes, mode="r")
-
-            (
-                content_items_file_names_tuple,
-                scripts_id_to_name,
-            ) = self.find_uuids_in_content_item(tar)
-            self.write_custom_content(
-                content_items_file_names_tuple, scripts_id_to_name
-            )
-            return True
+            api_response: HTTPResponse = demisto_client.generic_request_func(
+                self.client, "/content/bundle", "GET",  _preload_content=False,
+            )[0]
 
-        except ApiException as e:
-            self.handle_api_exception(e)
-            return False
-        except MaxRetryError as e:
-            self.handle_max_retry_error(e)
-            return False
         except Exception as e:
-            logger.info(f"Exception raised when fetching custom content:\n{e}")
-            return False
+            if isinstance(e, ApiException) and e.status == 401:
+                logger.error(f"Server authentication error: {e}\n"
+                             "Please verify that the required environment variables ('DEMISTO_API_KEY', or "
+                             "'DEMISTO_USERNAME' and 'DEMISTO_PASSWORD') are properly configured."
+                             )
```

Comment:
Because I have shared code for all of the exceptions, and instead of copy-pasting it for each `except` block, I just use a single `except` block, with `if` blocks for just for logginf different errors.

Full code:
```python
        except Exception as e:
            if isinstance(e, ApiException) and e.status == 401:
                logger.error(f"Server authentication error: {e}\n"
                             "Please verify that the required environment variables ('DEMISTO_API_KEY', or "
                             "'DEMISTO_USERNAME' and 'DEMISTO_PASSWORD') are properly configured."
                             )

            elif isinstance(e, MaxRetryError):
                logger.error(f"Failed connecting to server: {e}.\n"
                             "Please verify that the environment variable 'DEMISTO_BASE_URL' is properly configured, "
                             "and that the server is accessible.\n"
                             "If the server is using a self-signed certificate, try using the '--insecure' flag.")

            else:
                logger.error(f"Error while fetching custom content: {e}")

            logger.debug(traceback.format_exc())
            raise HandledError from e
```